Order state updated from `filling` to `open`

It appears its possible that the app receives a `offer` before it finished to submit the market order. Judging from the logs below, the order was set to `Filling` when the `RenewOffer` was received and afterwards set to `Open`. That unfortunately overwrites the execution price, thats why the app finally crashes because the order did not have an execution price.

```
[ERROR] TIME: 2024-05-15T18:26:50.552849Z r: Aborting after panic in task info: panicked at mobile/native/src/trade/position/mod.rs:88:59:
order to be filled
```


```
[INFO] TIME: 2024-05-15T18:26:45.468012Z r: Processing message from: 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156,kind: RenewOffer
[DEBUG] TIME: 2024-05-15T18:26:45.468842Z r: Received message from: 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156
[INFO] TIME: 2024-05-15T18:26:45.487135Z r: Automatically accepting renew offer channel_id: b6d22404bc4ef1e2c38654a982cced949bd5d4b3299b7954b1726af40337718d
[INFO] TIME: 2024-05-15T18:26:45.491164Z r: Received match from orderbook order_id: ba7d33e2-dbff-4025-899e-af926a0caee8
[INFO] TIME: 2024-05-15T18:26:45.496266Z r: Updated order state new_state: Filling,order_id: ba7d33e2-dbff-4025-899e-af926a0caee8
[WARN] TIME: 2024-05-15T18:26:45.497527Z r: Could not remove non-existing order order_id: f3512254-099b-4607-a74d-37af01fddb9a
[WARN] TIME: 2024-05-15T18:26:45.523602Z r: Could not remove non-existing order order_id: f4fc4404-3659-4b74-995d-37651b6946f0
[WARN] TIME: 2024-05-15T18:26:45.524633Z r: Could not remove non-existing order order_id: 4eadb96f-6999-4a63-b173-e7e101b4f4b3
[WARN] TIME: 2024-05-15T18:26:45.525520Z r: Could not remove non-existing order order_id: 9a4f1eb6-2ee7-49b4-8e03-cecf0460ccba
[INFO] TIME: 2024-05-15T18:26:45.541125Z r: Updated order state new_state: Open,order_id: ba7d33e2-dbff-4025-899e-af926a0caee8
```

--- a/mobile/native/src/trade/order/handler.rs
+++ b/mobile/native/src/trade/order/handler.rs
@@ -108,6 +108,7 @@ pub async fn submit_order_internal(
     let url = Url::parse(&url).expect("correct URL");
     let orderbook_client = OrderbookClient::new(url);
 
+    set_order_to_open_and_update_ui(order.id).map_err(SubmitOrderError::Storage)?;
     if let Err(err) = orderbook_client
         .post_new_market_order(order.clone().into(), channel_opening_params)
         .await

